internal/querynodev2/segments/collection.go

```diff
@@ -299,6 +299,18 @@ func NewCollectionWithoutSchema(collectionID int64, loadType querypb.LoadType) *
 	}
 }
 
+// new collection without segcore prepare
+// ONLY FOR TEST
+func NewCollectionWithoutSegcore(collectionID int64, schema *schemapb.CollectionSchema) *Collection {
```

Comment:
reminder: do we need to delete this

Reply:
All Collection struct field was private, so we must provide a special create function if some test in other package need use meta without segcore.